src/beacon-api/app/services/inlab_batch_service.py

```
@@ -0,0 +1,484 @@
+
+def _parse_dt(value: Optional[str]) -> Optional[datetime]:
+    """ISO string → aware datetime (UTC if naive), or None."""
+    if not value:
+        return None
+    dt = datetime.fromisoformat(value.replace("Z", "+00:00"))
+    if dt.tzinfo is None:
+        dt = dt.replace(tzinfo=timezone.utc)
+    return dt
+
+
+def _active_batch_for_device(db: Session, device_id: str, exclude_batch_id: UUID = None):
```

Ruff (0.15.12)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)